internal/dag/conditions.go

```diff
@@ -70,3 +70,22 @@ func headerConditions(conds []projcontour.Condition) []HeaderCondition {
 	}
 	return hc
 }
+
+func headerConditionsAreValid(conditions []projcontour.Condition) bool {
+	// Look for duplicate headers on conditions
+	// if found, set error condition on HTTPProxy
+	encountered := map[string]bool{}
+	for _, v := range conditions {
+		if v.Header == nil {
+			continue
+		}
+		switch {
+		case v.Header.Exact != "":
+			if encountered[v.Header.Name] {
```

Comment:
I think this needs to be a case insensitive comparison; this test case https://github.com/projectcontour/contour/blob/master/internal/dag/conditions_test.go#L196 should fail.